config/schemas/config.yaml

```diff
@@ -1014,6 +1014,11 @@ allOf:
               enabled:
                 type: boolean
                 const: true
+              buckets:
+                items:
+                  properties:
+                    destinationType:
+                      const: "swift"
```

Comment:
This will not capture the case fully, since if we have thanos with swift enabled, and we haven't set `destinationType` then it should default to swift.

We could potentially rework so the condition only looks for `destinationType: swift` for it to be simpler, but that is a different thing.

Reply:
So we should check, if any bucket has `destinationType: swift`, or if swift for thanos is enabled and we haven't specified `destinationType` for the bucket (and the same thing for harbor) - and if any of these are true, we should perform the check for `networkPolicies.rclone.sync.objectStorageSwift`?

Reply:
And if that is the default behavior we want, then we would need to incorporate a lot into the [conditional set-me](https://github.com/elastisys/compliantkubernetes-apps/blob/main/config/sc-config.yaml#L1304).

Reply:
Yes, that is true. Though the new schema addition is :ok_hand: I didn't know it was possible to express it that succinct.

Reply:
It would be to expand to something like this `.harbor.persistence.type == "swift" and (.objectStorage.sync.buckets[] | select(.source == "*harbor*") | .destinationType != {})` but I'm not sure we _want_ to expand the set-mes, we should probably work towards them being expressed in the schema only.

Reply:
@Ajarmar Do you have any thoughts on this?

Reply:
I have a monstrosity of a conditional set-me that seems to be working. However in making it I have found some issues with the schema changes I made that I'll have to try and sort out.

Reply:
> @Ajarmar Do you have any thoughts on this?

It would be nice to only have to specify the condition in one place rather than two as we do now (config and schema), but this seems like a bigger initiative than what should be done in this PR. For instance, a schema validation failure does not halt an automatic apps upgrade, but an unset set-me does, so this is something that would be missed if we were to only express the condition in the schema right now.